guardrails/src/adapters/llm/cube_llm.py

```
@@ -0,0 +1,211 @@
+from typing import Any, Dict, List, Optional
+
+from langchain_core.messages import BaseMessage
+from langchain_core.outputs import ChatResult
+from langchain_openai import ChatOpenAI
+from nemoguardrails.context import generation_options_var
+
+logger = logging.getLogger(__name__)
+
+
+class CubeLLM(ChatOpenAI):
+    _config_headers: Dict[str, str] = {}
```

Ruff (0.15.1)

[warning] Mutable default value for class attribute

(RUF012)